app/com/linkedin/drelephant/clients/azkaban/AzkabanWorkflowClient.java

```diff
@@ -394,24 +394,25 @@ private String decodeHeadlessChallenge(String encodedPassword, File _privateKey)
       JSONObject jsonObject = fetchJson(urlParameters, _workflowExecutionUrl);
       JSONArray jobs = jsonObject.getJSONArray("nodes");
       Map<String, String> jobMap = new HashMap<String, String>();
-      for (int i = 0; i < jobs.length(); i++) {
-        JSONObject job = jobs.getJSONObject(i);
-        jobMap.put(job.get("id").toString(), job.get("status").toString());
-        if (!job.isNull("nodes")) {
-          JSONArray internalJobs = job.getJSONArray("nodes");
-          for (int j = 0; j < internalJobs.length(); j++) {
-            JSONObject internalJob = internalJobs.getJSONObject(j);
-            jobMap.put(internalJob.get("id").toString(), internalJob.get("status").toString());
-          }
-        }
-      }
+      addJobStatusForFlow(jobMap, jobs);
       return jobMap;
     } catch (JSONException e) {
       logger.error("Error in parsing azkaban output ", e);
     }
     return null;
   }
 
+  public void addJobStatusForFlow(Map<String, String> jobMap, JSONArray jobs) throws JSONException {
```

Comment:
Should you check if job is null or not before starting loop

Reply:
Added a check.